lib/jsdom/living/nodes/HTMLElement-impl.js

```diff
@@ -35,6 +42,54 @@ class HTMLElementImpl extends ElementImpl {
     }
   }
 
+  // https://html.spec.whatwg.org/multipage/custom-elements.html#attached-internals
+  attachInternals() {
+    if (this._isValue !== null) {
+      throw DOMException.create(this._globalObject, [
+        "Unable to attach ElementInternals to a customized built-in element.",
+        "NotSupportedError"
+      ]);
+    }
+
+    const definition = lookupCEDefinition(this.ownerDocument, this.namespaceURI, this.localName, null);
+
+    if (definition === null) {
+      throw DOMException.create(this._globalObject, [
+        "Unable to attach ElementInternals to non-custom elements.",
+        "NotSupportedError"
+      ]);
+    }
+
+    if (definition.disableInternals === true) {
+      throw DOMException.create(this._globalObject, [
+        "ElementInternals is disabled by disabledFeature static field.",
+        "NotSupportedError"
+      ]);
+    }
+
+    if (this._attachedInternals !== null) {
+      throw DOMException.create(this._globalObject, [
+        "ElementInternals for the specified element was already attached.",
+        "NotSupportedError"
+      ]);
+    }
+
+    if (!["precustomized", "custom"].includes(this._ceState)) {
+      throw DOMException.create(this._globalObject, [
+        "The attachInternals() function cannot be called prior to the execution of the custom element constructor.",
+        "NotSupportedError"
+      ]);
+    }
+
+    this._attachedInternals = ElementInternals.new(this._globalObject);
+    const attachedInternalsImpl = wrapperForImpl(this._attachedInternals);
+
+    // https://html.spec.whatwg.org/multipage/custom-elements.html#internals-target
+    attachedInternalsImpl._targetElement = this;
+
+    return attachedInternalsImpl;
```

Comment:
I'm not sure whether there should use  `ElementInternals.new` and `wrapperForImpl`, or use other method.